openedx/features/course_experience/views/course_home_messages.py

```diff
@@ -122,81 +128,100 @@ def _register_course_home_messages(request, course_id, user_access, course_start
+        CourseGoalOption.EXPLORE.value,
+    ]
+    for goal_key in goal_options:
+        goal_text = get_goal_text(goal_key)
+        goal_choices_html += HTML(
+            '{initial_tag}{goal_text}{closing_tag}'
         ).format(
             initial_tag=HTML(
-                '<div tabindex="0" aria-label="{aria_label_choice}" class="goal-option dismissible" '
+                '<div tabindex="0" aria-label="{aria_label_choice}" class="goal-option {col_sel} btn" '
                 'data-choice="{goal_key}">'
             ).format(
-                goal_key=CourseGoalOption.UNSURE.value,
-                aria_label_choice=Text(_("Set goal to: {choice}")).format(
-                    choice=get_goal_text(CourseGoalOption.UNSURE.value)
+                goal_key=goal_key,
+                aria_label_choice=Text(_("Set goal to: {goal_text}")).format(
+                    goal_text=Text(_(goal_text))
                 ),
+                col_sel='col-' + str(int(math.floor(12 / len(goal_options))))
             ),
-            choice=Text(_('{choice}')).format(
-                choice=get_goal_text(CourseGoalOption.UNSURE.value),
-            ),
-            closing_tag=HTML('</div>'),
+            goal_text=goal_text,
+            closing_tag=HTML('</div>')
         )
 
-        # Add the option to set a goal to earn a certificate,
-        # complete the course or explore the course
-        goal_options = [CourseGoalOption.CERTIFY.value, CourseGoalOption.COMPLETE.value, CourseGoalOption.EXPLORE.value]
-        for goal_key in goal_options:
-            goal_text = get_goal_text(goal_key)
-            goal_choices_html += HTML(
-                '{initial_tag}{goal_text}{closing_tag}'
-            ).format(
-                initial_tag=HTML(
-                    '<div tabindex="0" aria-label="{aria_label_choice}" class="goal-option {col_sel} btn" '
-                    'data-choice="{goal_key}">'
-                ).format(
-                    goal_key=goal_key,
-                    aria_label_choice=Text(_("Set goal to: {goal_text}")).format(
-                        goal_text=Text(_(goal_text))
-                    ),
-                    col_sel='col-' + str(int(math.floor(12 / len(goal_options))))
-                ),
-                goal_text=goal_text,
-                closing_tag=HTML('</div>')
-            )
-
-        CourseHomeMessages.register_info_message(
-            request,
-            HTML('{goal_choices_html}{closing_tag}').format(
-                goal_choices_html=goal_choices_html,
-                closing_tag=HTML('</div>')
-            ),
-            title=Text(_('Welcome to {course_display_name}')).format(
-                course_display_name=course.display_name
-            )
+    CourseHomeMessages.register_info_message(
+        request,
+        HTML('{goal_choices_html}{closing_tag}').format(
+            goal_choices_html=goal_choices_html,
+            closing_tag=HTML('</div>')
```

Comment:
I "think" the closing tag is already included in `goal_choices_html` above where it belongs.  So maybe this could be?:
```
CourseHomeMessages.register_info_message(
    request,
    goal_choices_html,
    title=Text(_('Welcome to {course_display_name}')).format(
        course_display_name=course.display_name
    )
```

Reply:
I believe that this is right, @robrap. I've made the change and we'll see how it looks.